src/userLayout/RoomDetails/RoomDetails.tsx

```diff
@@ -0,0 +1,501 @@
+                        color: "#152C5B",
+                        fontSize: "38px",
+                        height: "38px",
+                      }}
+                    />
+                    <Box style={{ color: "#B0B0B0" }}> Mbp/s</Box>
+                  </Box>
+                  <Box sx={{ marginLeft: "2rem" }}>
+                    <AcUnitIcon
+                      style={{
+                        color: "#152C5B",
+                        fontSize: "38px",
+                        height: "38px",
+                      }}
+                    />
+                    <Box style={{ color: "#B0B0B0" }}>Unit Ready</Box>
+                  </Box>
+
+                  <Box sx={{ marginLeft: "2rem" }}>
+                    <BluetoothIcon
+                      style={{
+                        color: "#152C5B",
+                        fontSize: "38px",
+                        height: "38px",
+                      }}
+                    />
+                    <Box style={{ color: "#B0B0B0" }}> Bluetooth</Box>
+                  </Box>
+
+                  <Box sx={{ marginLeft: "2rem" }}>
+                    <TvIcon
+                      style={{
+                        color: "#152C5B",
+                        fontSize: "38px",
+                        height: "38px",
+                      }}
+                    />
+                    <Box style={{ color: "#B0B0B0" }}> television</Box>
+                  </Box>
+                </Box>
+              </Grid>
+
+              {/* border */}
+              <Grid item xs={12} md={12} lg={4}>
+                <Box
+                  sx={{
+                    border: "1px solid #E5E5E5",
+                    padding: "5rem",
+                    borderRadius: "1rem",
+                  }}
+                >
+                  <Typography
+                    variant="h5"
+                    color="secondary"
+                    sx={{ color: "#1a237e", marginTop: "2" }}
+                  >
+                    Start Booking
+                  </Typography>
+
+                  <Typography
+                    variant="body1"
+                    sx={{
+                      color: "#1ABC9C",
+                      fontSize: "36px",
+                      fontWeight: "300",
+                      marginTop: "1rem",
+                    }}
+                  >
+                    {price}
+                    <span style={{ color: "#B0B0B0", marginLeft: "0.5rem" }}>
+                      per night
+                    </span>
+                  </Typography>
+                  <Calendar {...{ selectedDateRange, setSelectedDateRange }} />
+
+                  {/* btn */}
+
+                  <Box
+                    sx={{
+                      display: "flex",
+                      justifyContent: "center",
+                      marginTop: "3rem",
+                    }}
+                  >
+                    <Link to="/checkout" state={bookingId}>
```

Comment:
We shouldn't be nesting interactive elements like buttons and anchor elements.
We have two options for navigation either using a Link component or programmatic  navigation (performing some code and then navigate ) using useNavigate hook.

We use the Link component when we want to perform a navigation action with nothing else.
We use useNavigate hook when we need to perform some code  before navigation, like in our example we need to create a booking and then navigate.